lldb/include/lldb/Breakpoint/WatchpointResourceList.h

```diff
@@ -0,0 +1,146 @@
+//===-- WatchpointResourceList.h --------------------------------*- C++ -*-===//
```

Comment:
I think it's the same except for the obvious BreakpointSite/WatchpointResource and BreakpointLocation/Watchpoint.  It's a little tricky reading through BreakpointSiteList.h to tell if all of the arguments refer to BreakpointLocations (they do it by id values) or if they might refer to Breakpoints - they seem to all use the same `break_id_t`?  (and `break_id_t` is also used for the BreakpointSite numbers); sometimes the arguments are called `bp_id` and sometimes they're called `BreakID` but I THINK they're always referring to BreakpointLocations.

It'd be nice if BreakpointSite and WatchpointResource could inherit from the same base class and have the StopPointList work in terms of the base class methods, but I think there's enough difference between these two that this might not work very cleanly.

Maybe a templated StopPointList could work though.  I'll need to look at that.